### DISCUSSION TITLE
Experimental-Ags branch Q&A

### DISCUSSION THREAD

**Discussion by JaKooLit** [JaKooLit/Hyprland-Dots#239] [Ideas]
@SherLock707 @JohnRTitor 

so I will try now this project and possibly contribute and start tweaking..

So I download the branch

installed ags with gnome-bluetooth-3.0
Place the ags folder in ~/.config

Edit startup for starting ags

placed this on keybinds...

bind = $mainMod, A, exec, pkill rofi || true && ags -t 'overview'

not launching.. am I missing more?

**Comment by SherLock707** [comment]
Is ags running in background?

**Reply by SherLock707** [reply]
Also, whats gnome-bluetooth-3.0?

**Comment by JohnRTitor** [comment]
Yeah you need to exec once it.

**Comment by SherLock707** [comment]
After `ags &` try `ags -t 'overview'` in terminal.

**Comment by JaKooLit** [comment]
> Also, whats gnome-bluetooth-3.0?

when I tried launching ags on tty, it comes back with error gnome-bluetooth-3.0 missing
apparently its one of the dependencies

https://aylur.github.io/ags-docs/config/installation/

**Reply by SherLock707** [reply]
Oh, I need to check my installation (I don't have access to PC at the moment, I'll have in 5-6 hrs)

**Reply by SherLock707** [reply]
Is it launching after you install the deps?

**Reply by JaKooLit** [reply]
no problem bro, just try later.. and also, try to update to v.0.40.0

**Reply by SherLock707** [reply]
I'm running latest hyprland, not sure about version, I'll check.

**Comment by JaKooLit** [comment]
> After `ags &` try `ags -t 'overview'` in terminal.

![swappy-20240506-133528](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/458bc85e-5740-417f-9266-7d359a6a9260)

nothing happened... perhaps because I am using now hyprland v0.40.0?

**Reply by SherLock707** [reply]
Run ags debug window, `ags --help` there was a flag for debug.

**Reply by JohnRTitor** [reply]
It's working fine on Hyprland v40 on NixOS. Instead of manual changes testing, I'd recommend cherry-picking the commits from experimental-ags branch, so that nothing is left out.

**Reply by SherLock707** [reply]
Lastest (non-git) hyprland on Arch here. Works fine..

**Comment by JaKooLit** [comment]
> Oh, I need to check my installation (I don't have access to PC at the moment, I'll have in 5-6 hrs)



> Is it launching after you install the deps?

yeah no issue after launching.. it seems running in background.. just not launching the overview.. see my screenshot

**Reply by SherLock707** [reply]
Kill or quit ags.
Open two terminals 
Run `ags` in first
Run `ags -t 'overview'` in second.

When you launch widget, do you see any error in first terminal?

**Comment by JaKooLit** [comment]
> Kill or quit ags. Open two terminals Run `ags` in first Run `ags -t 'overview'` in second.
> 
> When you launch widget, do you see any error in first terminal?

![Screenshot_06-May_13-45-01_653](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/3f775d4c-8550-4804-865a-930ccca1d09a)
nothing happened...

newly reboot

**Reply by SherLock707** [reply]
Right window says its already running, kill that one and launch new and don't send to background so that you can see the cli output.

**Reply by JaKooLit** [reply]
duh.. im dumb

`┌─[ja@Arch-G15-HyprLanD] - [~] - [Mon May 06, 14:00]
└─[$] <> ags              

(ags:3935): Gtk-WARNING **: 14:00:36.122: Theme parsing error: gtk.css:6616:68: Invalid name of pseudo-class

(ags:3935): Gjs-WARNING **: 14:00:36.144: JS ERROR: ImportError: Unable to load file from: resource:///com/github/Aylur/ags/service/applications.js (The resource at “/com/github/Aylur/ags/service/applications.js” does not exist)
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34`

**Reply by JohnRTitor** [reply]
lmao

**Reply by SherLock707** [reply]
Was there some ags js repo change recently?🤔 May be ours is cached somewhere and since you are running new, its not able to pull? 

Any luck googling this?

**Comment by JaKooLit** [comment]
well same result with ags-git

**Comment by JaKooLit** [comment]
![Screenshot_06-May_01-20-16_3350](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/361f3165-3d84-445b-a172-b71c4ebaf513)
![Screenshot_06-May_01-21-16_9376](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/eb83e171-8c6d-4f37-b2d3-ac57f1797cc9)
![Screenshot_06-May_01-21-03_23564](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/54e9658c-4178-41ed-a78c-4d8bdc324ed0)

its running on Fedora lmao

**Reply by SherLock707** [reply]
Its running on arch for me👀

**Reply by SherLock707** [reply]
I think I have ags from Chaotic AUR.

**Comment by SherLock707** [comment]
We need to figure out how to make this js file pull from repo to local js files may be.. so that this does not happen..

**Reply by SherLock707** [reply]
Is there a difference between pkg version in arch and fedora for ags? Also AUR?

**Comment by JaKooLit** [comment]
I think its my Arch on my laptop is not healthy.. I start to get lots of kernel panic latety.. Ill try on my mini PC later on...

I think its time to completely reinstall my Arch on my laptop after almost 3.5 yrs of no reformat 😀

**Comment by SherLock707** [comment]
> I think its my Arch on my laptop is not healthy.. I start to get lots of kernel panic latety.. Ill try on my mini PC later on...
> 
> I think its time to completely reinstall my Arch on my laptop after almost 3.5 yrs of no reformat 😀

If you are reinstalling arch, Try Garuda-linux hyprland edition. Its basically Arch with btrfs snapshot setup and nicely pre installed and configured tools out of the box.
Also chaotic AUR.

**Reply by JaKooLit** [reply]
> > I think its my Arch on my laptop is not healthy.. I start to get lots of kernel panic latety.. Ill try on my mini PC later on...
> > 
> > I think its time to completely reinstall my Arch on my laptop after almost 3.5 yrs of no reformat 😀
> 
> If you are reinstalling arch, Try Garuda-linux hyprland edition. Its basically Arch with btrfs snapshot setup and nicely pre installed and configured tools out of the box.
> Also chaotic AUR.

Thanks but no thanks... I dont have desire nor love for any remastered distros 😁 as my main distro

Ive been using btrs with snapshot already for time already so its not a new to me

**Reply by SherLock707** [reply]
Oh cool, just suggested since saves time in installation.

**Comment by JaKooLit** [comment]
Oh well.. its definitely my laptop 🤣
![Screenshot_06-May_16-13-24_8558](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/dac294ed-905a-4841-8815-946b41a364e8)

**Reply by SherLock707** [reply]
Original repo has ags bar + notification pannel + OSD and other stuff in single ags package(just needs pywal integration) .. guess which is the end goal for this if you liked ags..

**Comment by JaKooLit** [comment]
https://github.com/JaKooLit/Hyprland-Dots/commit/e1a22b19efdf542480868f1c250f11ace3a275c1

Please check my latest commit.. still need some proper tweaking..

But this should integrate well with Dark-Light.sh... for now.... 

:)

Let me know

**Reply by SherLock707** [reply]
Will do.

**Comment by SherLock707** [comment]
@JaKooLit , off topic: waybar-cava module which is piped via cava in a shell script. When you are playing some audio (module is actively displaying bars) any of the other modules tooltip works for you (when you hover)?

**Reply by JaKooLit** [reply]
oh... wow.. didnt even noticed that until now... seems the cava-waybar module / script messed up the tooltip..

but if differenr waybar config where no cava, tooltip works fine

**Reply by SherLock707** [reply]
I think it is because that cava scripts pipes too much, too fast.

**Comment by JaKooLit** [comment]
I have now merge to development branch

https://github.com/JaKooLit/Hyprland-Dots/pull/241

**Reply by SherLock707** [reply]
Awesome!
